controlplane/main.go

```diff
@@ -129,20 +132,32 @@ func main() {
 	}
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:             scheme,
-		MetricsBindAddress: metricsBindAddr,
-		LeaderElection:     enableLeaderElection,
-		LeaderElectionID:   "rke2-controlplane-manager-leader-election-capi",
-		LeaseDuration:      &leaderElectionLeaseDuration,
-		RenewDeadline:      &leaderElectionRenewDeadline,
-		RetryPeriod:        &leaderElectionRetryPeriod,
-		SyncPeriod:         &syncPeriod,
-		ClientDisableCacheFor: []client.Object{
-			&corev1.ConfigMap{},
-			&corev1.Secret{},
+		Scheme: scheme,
+		Metrics: metricsserver.Options{
```

Comment:
it is suggested to use a new flag `--diagnostics-address` instead of `metrics-bind-address` (xref: https://main.cluster-api.sigs.k8s.io/developer/providers/migrations/v1.5-to-v1.6#suggested-changes-for-providers), do we want to incorporate that change in the same PR?

Reply:
No objections from me 👍🏼

Reply:
Added #216 which has also been added to the v0.3.0 milestone.